### PR TITLE
[FE-28] Implemented Live broadcast feature

### PR DESCRIPTION
**Pull request by kodeman2**
# Fixes Issue/Linear Ticket

( This could be an existing issue or a linear ticket )

**My PR Closes #issue_number_here** or
**My PR Fixes ID-Team-plug-space/issue/FE-28/implement-video-Broadcast**
Linearlink: https://linear.app/team-plug-space/issue/FE-28/implement-video-broadcast 

# Changes proposed

## What were you told to do?

Implement Video Broadcast using agora ui engine

## What did you do?

i implemented video broadcast feature using agora UI engine
also added the feature livebroadcast button to the sideBar

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/ Videos
#sceenshot
zurifeature
#new changes screenshot
newzurifeat

 

#video

https://user-images.githubusercontent.com/97587019/203947310-9b37cec4-4e8f-4d82-baf1-41a8b006c06f.mp4

#new changes video

https://user-images.githubusercontent.com/97587019/204111198-a8711864-94e9-42c4-8c86-5f47fbdd59c2.mp4

### CODE REVIEW

**Comment by billmal071** [comment]
Didn't someone already implement this.

**Comment by kodeman2** [comment]
The live broadcast feature is a different ticket from the video call feature sir

**Comment by kodeman2** [comment]
> Didn't someone already implement this.



> Didn't someone already implement this.

The live broadcast feature is a different ticket from the video call feature sir

**Comment by billmal071** [comment]
> > Didn't someone already implement this.
> 
> > Didn't someone already implement this.
> 
> The live broadcast feature is a different ticket from the video call feature sir

Alright, but that svg is to big, I believe there should be one you can get from react-icons.

Also, look at slacks method for videos and implement it like that, the icons don't look good also.

**Comment by kodeman2** [comment]
> > > Didn't someone already implement this.
> > 
> > 
> > > Didn't someone already implement this.
> > 
> > 
> > The live broadcast feature is a different ticket from the video call feature sir
> 
> Alright, but that svg is to big, I believe there should be one you can get from react-icons.
> 
> Also, look at slacks method for videos and implement it like that, the icons don't look good also.

kindly help review, thanks

File: packages/main/src/components/media-chat/LiveBroadcast.jsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+const LiveBroadcast = () => {
+  const [videocall, setVideocall] = useState(true);
+  const [isHost, setHost] = useState(true);
+
+  const rtcProps = {
+    appId: "9e95b753731148ff9f6ab866502fe3df",
+    channel: "live",
+    token:
+      "007eJxTYChtKGew0hY63SLCajbvL4Nj23lpM3Mb09Qzwv0eczyPNikwWKZamiaZmxqbGxsamlikpVmmmSUmWZiZmRoYpaUap6RZZzYlNwQyMtyaK8vEyACBID4LQ05mWSoDAwD7whta", //add your token if using app in secured mode
```
Comment (ickynavigator) [inline packages/main/src/components/media-chat/LiveBroadcast.jsx:13]: stop committing api tokens
another member of your team has already changed this to use an ENV instead

File: packages/main/src/components/protected/sidebar/components/Sidebar.jsx [hunk 2]
```diff
@@ -4,6 +4,8 @@ import { useTranslation } from "react-i18next";
 
 import threadIcon from "../assets/icons/thread-icon.svg";
 import dmIcon from "../assets/icons/dm-icon.svg";
+import liveicon from "../assets/icons/newlive.svg";
+
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/Sidebar.jsx:8]: reason for this space?

File: packages/main/src/components/protected/sidebar/components/Sidebar.jsx [hunk 3]
```diff
@@ -230,6 +238,8 @@ const Sidebar = props => {
 
 export default Sidebar;
 
+/*
```
Comment (ickynavigator) [inline packages/main/src/components/protected/sidebar/components/Sidebar.jsx:241]: reason for this comment?

**Review by ickynavigator — CHANGES_REQUESTED** [review]

File: packages/main/src/components/protected/sidebar/components/Sidebar.jsx (hunk 3 above)
Comment (kodeman2) [reply packages/main/src/components/protected/sidebar/components/Sidebar.jsx]: Removed unnecessary comments

File: packages/main/src/components/protected/sidebar/components/Sidebar.jsx (hunk 2 above)
Comment (kodeman2) [reply packages/main/src/components/protected/sidebar/components/Sidebar.jsx]: I've removed the space .. thanks

File: packages/main/src/components/media-chat/LiveBroadcast.jsx (hunk 1 above)
Comment (kodeman2) [reply packages/main/src/components/media-chat/LiveBroadcast.jsx]: I've stored them in an env file

File: example.env [hunk 4]
```diff
@@ -1,2 +1,3 @@
 REACT_APP_AGORA_TOKEN=007e*****
 REACT_APP_AGORA_APP_ID=***************
+
```
Comment (ickynavigator) [inline example.env:3]: this space isn't necessary

Comment (kodeman2) [reply example.env]: Removed already sir

Comment (kodeman2) [reply example.env]: Removed already sir

**Comment by kodeman2** [comment]
@billmal071 @ickynavigator  I'm done with the changes requested...thanks

**Comment by JulianaSau** [comment]
@kodeman2 include a screenshot of how the icon on the sidebar looks now. And if you're no longer using the SVG remove it from your commits

**Comment by kodeman2** [comment]
> 

done i've added a new screenshot like you requested
newscreenshot

**Comment by kodeman2** [comment]
@JulianaSau  i've done what you requested ma

**Comment by kodeman2** [comment]
@JulianaSau Thanks🙌

**Comment by JulianaSau** [comment]
@kodeman2 please be in contact with mayowa to give him the environment variable values required for the broadcast to run successfully

**Comment by kodeman2** [comment]
Alright.. I'd do that

**Pull request merged**